Author: Artem Slepnev "Shellgen"
Description: https - nossl support
Ver: elinks 0.14.3
Date: 2022-01-14 19:40 UTC
==============================================================
--- a/src/protocol/protocol.c
+++ b/src/protocol/protocol.c
@@ -67,7 +67,7 @@
 	{ "ftp",	  21, ftp_protocol_handler,		1, 1, 0, 0, 0 },
 	{ "gopher",	  70, gopher_protocol_handler,		1, 1, 0, 0, 1 },
 	{ "http",	  80, http_protocol_handler,		1, 1, 0, 0, 1 },
-	{ "https",	 443, https_protocol_handler,		1, 1, 0, 1, 1 },
+	{ "https",	 443, http_protocol_handler,		1, 1, 0, 0, 1 },
 	{ "javascript",	   0, NULL,				0, 0, 1, 0, 1 },
 	{ "mailcap",	   0, mailcap_protocol_handler,		0, 0, 1, 0, 0 },
 	{ "news",	   0, news_protocol_handler,		0, 0, 1, 0, 1 },
