Author: Artem Slepnev "Shellgen"
Description: remove check architecture (support x32)
Ver: boost-1.58.0 - boost-1.72.0-r1  ( <=boost-1.58.0 )
Date: 2018-06-24 07:56 UTC
==============================================================
--- a/Jamroot
+++ b/Jamroot
@@ -136,7 +136,6 @@
 
 boostcpp.set-version $(BOOST_VERSION) ;
 
-use-project /boost/architecture : libs/config/checks/architecture ;
 
 local all-headers =
     [ MATCH .*libs/(.*)/include/boost : [ glob libs/*/include/boost ] ] ;
@@ -164,8 +163,6 @@
 project boost
     : requirements <include>.
 
-      [ boostcpp.architecture ]
-      [ boostcpp.address-model ]
 
       # Disable auto-linking for all targets here, primarily because it caused
       # troubles with V2.
