Author: Artjom Slepnjov "Shellgen"
Description: html page - servername remove
Ver: webfs-1.21
Date: 2023-06-18 15:14 UTC
==============================================================
--- ls.c
+++ ls.c
@@ -18,7 +18,8 @@
 #include "httpd.h"
 
 #define LS_ALLOC_SIZE (4 * 4096)
-#define HOMEPAGE "http://bytesex.org/webfs.html"
+#define HOMEPAGE "License: GPL-2"
+#define SERVERNAME "Author: Gerd Knorr"
 
 #ifdef USE_THREADS
 static pthread_mutex_t lock_dircache = PTHREAD_MUTEX_INITIALIZER;
@@ -265,10 +266,10 @@
     len  = 0;
 
     len += sprintf(buf+len,
-		   "<head><title>%s:%d%s</title></head>\n"
+		   "<head><title>%s%s</title></head>\n"
 		   "<body bgcolor=white text=black link=darkblue vlink=firebrick alink=red>\n"
 		   "<h1>listing: \n",
-		   hostname,tcp_port,path);
+		   hostname,path);
 
     h1 = path, h2 = path+1;
     for (;;) {
@@ -374,7 +375,7 @@
 		   "</pre><hr noshade size=1>\n"
 		   "<small><a href=\"%s\">%s</a> &nbsp; %s</small>\n"
 		   "</body>\n",
-		   HOMEPAGE,server_name,line);
+		   HOMEPAGE,SERVERNAME,line);
     for (i = 0; i < count; i++)
 	free(files[i]);
     if (count)
